cargo-dist/src/host.rs

```diff
@@ -269,16 +273,29 @@ fn announce_hosting(_dist: &DistGraph, manifest: &DistManifest, abyss: &Gazenot)
 
 pub(crate) fn select_hosting(
     workspace: &WorkspaceInfo,
+    announcing: &AnnouncementTag,
     hosting: Option<Vec<HostingStyle>>,
     ci: Option<&[CiStyle]>,
 ) -> Option<HostingInfo> {
+    let package_list = announcing
+        .rust_releases
+        .iter()
+        .map(|(idx, _)| *idx)
+        .collect::<Vec<_>>();
     // Either use the explicit one, or default to the CI provider's native solution
     let hosting_providers = hosting
         .clone()
         .or_else(|| Some(vec![ci.as_ref()?.first()?.native_hosting()?]))?;
-    let repo_url = workspace.web_url().unwrap_or_default()?;
+    // Check that there's a consistent repository URL, and if not, warn
+    if let Err(warning) = workspace.repository_url(Some(&package_list)) {
+        let report = miette::Report::new(warning);
```

Comment:
Curious why we need to construct a miette report here if we're just passing it to a debug string - better formatting?

Reply:
warn!s are user-facing, and yeah wrapping in a Report is (and then debug formatting that report) is the ritual to get the Good miette output.